scipy/optimize/_zeros_py.py

```diff
@@ -1388,3 +1388,278 @@ def toms748(f, a, b, args=(), k=1,
+        a = xt
+        fa = ft
+
+        # set r so that f(r) is the minimum magnitude of f(a) and f(b)
+        fa_is_smaller = np.abs(fa) < np.abs(fb)
+        r = np.choose(fa_is_smaller, [b, a])
+        fm = np.choose(fa_is_smaller, [fb, fa])
+
+        tol = 2 * xatol * np.abs(r) + xrtol
+        tlim = tol / np.abs(b - c)
+        terminate = np.logical_or(terminate,
+                                  np.logical_or(fm == 0, tlim > 0.5))
+
+        # check convergence here, update flag if so
+        if np.all(terminate):
+            flag.fill(_ECONVERGED)
+            break
+
+        iterations += 1 - terminate
+
+        # Figure out values xi and phi
+        # to determine which method we should use next
+        xi = (a - b) / (c - b)
+        phi = (fa - fb) / (fc - fb)
+        iqi = np.logical_and(phi**2 < xi, (1 - phi) ** 2 < 1 - xi)
+
+        if iqi.all():
+            t = np.full(shape, 0.5)
+            a2, b2, c2, fa2, fb2, fc2 = (
+                a[iqi],
+                b[iqi],
+                c[iqi],
+                fa[iqi],
+                fb[iqi],
+                fc[iqi],
+            )
+            t[iqi] = fa2 / (fb2 - fa2) * fc2 / (fb2 - fc2) + (c2 - a2) / (
+                b2 - a2
+            ) * fa2 / (fc2 - fa2) * fb2 / (fc2 - fb2)
+        else:
+            # bisection
+            t = 0.5
+
+        # limit to the range (tlim, 1-tlim)
+        t = np.minimum(1 - tlim, np.maximum(tlim, t))
+
+        intermediate_result = OptimizeResult(x=r, fun=ft)
+        if _call_callback_maybe_halt(callback, intermediate_result):
+            break
+
+    # reshape outputs
+    r = np.reshape(r, intermediate_shape)
+    ft = np.reshape(ft, intermediate_shape)
+    funcalls = np.reshape(funcalls, intermediate_shape)
+    a = np.reshape(a, intermediate_shape)
+    b = np.reshape(b, intermediate_shape)
+    fa = np.reshape(fa, intermediate_shape)
+    fb = np.reshape(fb, intermediate_shape)
+    iterations = np.reshape(iterations, intermediate_shape)
+
+    # checks for convergence and conditions
+    if iterations.any() >= maxiter:
+        flag.fill(_ECONVERR)
+    elif not (np.sign(fa) * np.sign(fb) <= 0).all():
+        flag.fill(_ESIGNERR)
+    elif not ((np.isfinite(a).all() and np.isreal(a).all()) and
+              (np.isfinite(b).all() and np.isreal(b).all())):
+        flag.fill(_EVALUEERR)
+    elif not (np.isreal(fa).all() and np.isreal(fb).all()):
+        flag.fill(_EVALUEERR)
+    else:
+        flag.fill(_ECONVERGED)
+
+    # iterate here to create the message since the scalar case has dimension ()
+    message_flag = flag.flatten()
+    message = np.reshape([flag_map[i] for i in message_flag],
+                         intermediate_shape)
+    flag = np.reshape(flag, intermediate_shape)
+
+    result = RootResults(root=r,
+                         function_calls=funcalls,
+                         iterations=iterations,
+                         flag=1)  # place holder
```

Comment:
As a matter of style:
```suggestion
                         flag=None)  # place holder
```

Reply:
Got it, had to modify RootResults to an f-string to accept the `None` but hopefully that's okay.